pkg/apparmor/internal/supported/supported.go

```diff
@@ -0,0 +1,89 @@
+package supported
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/containers/storage/pkg/unshare"
+	runcaa "github.com/opencontainers/runc/libcontainer/apparmor"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+type ApparmorVerifier struct {
+	impl verifierImpl
+}
+
+func NewAppArmorVerifier() *ApparmorVerifier {
+	return &ApparmorVerifier{impl: &defaultVerifier{}}
+}
+
+// IsSupported returns nil if AppAmor is supported by the host system,
+// otherwise an error
```

Comment:
Note that we're looking up the binary twice which is not a huge cost but can add up.

Reply:
Yep, enhanced the comment to mention the error cases.